### DISCUSSION TITLE
Расчёт значений в полях, кодом из других полей.

### DISCUSSION THREAD

**Discussion by Ekhorikov2** [totumonline/totum-ru-issues-and-discussions#282] [Как это сделать?] [answered]
Всем привет.
Такая задача:
Есть расчётная таблица в цикле (выполняет функцию проекта, для расчёта стоимости оборудования и монтажа этого оборудования, для заказчика), в ней стандартно товары, кол-во, закупочная цена, себестоимость, прибыль и сумма.

![image](https://user-images.githubusercontent.com/112487577/205626758-1a32da79-4021-4bfd-b6d4-51629680f267.png)

В полях настроены коды:
**Количество:** нет
**Закупочная:** нет
**Себестоимость:** = : math` #kolichestvo * #zakupochnaya `
**Цена:** = : math` #zakupochnaya / (1 - (#pribylngost / 100)) `
**Прибыльность%:** нет
**Прибыль:** = : math` #summa - #sebestoimost `
**Сумма:** = : math` #cena * #kolichestvo `
Всё вроде бы ОК, заполняем кол-во, закупочную цену полученную у поставщика и % прибыльности, а остальные поля рассчитываются кодом. Но вот в реальной работе, такой таблицей пользоваться оказалось неудобно! Бывают случаи, когда нужно подогнать продажную стоимость в ручную (с пересчётом других полей), или сумму по товару (так же с пересчётом других полей)
Так вот вопрос в том, что я не могу сообразить как это сделать, т.к. если задать расчёт из другого поля, кодом, то тогда это поле выдаст ошибку расчётного значения.
Подскажите, как правильно задать коды, чтоб корректно работал пересчёт полей, при ручном изменении любого из трёх полей (ЦЕНА, ПРИБЫЛЬНОСТЬ и СУММА)? 
Заранее спасибо!

**Comment by ttm-alexeypolunin** [comment, marked as answer]
**Варианты:**

1. При добавлении у вас одно ведущее поле, которое он заполняет, а остальные рассчитываются. 

1.a. После добавления делаете код-действия, который при изменении сетит в стартовое поле и делает clear изменяемому.
1.a. У вас коды только при добавлении и вы при изменеии в одном поле и сетите два других
